pkg/config/model/viper.go

```diff
@@ -670,6 +702,60 @@ func (c *safeConfig) AddConfigPath(in string) {
 	c.Viper.AddConfigPath(in)
 }
 
+// AddExtraConfigPaths allows adding additional configuration files
+// which will be merged into the main configuration during the ReadInConfig call.
+// The algorithm used to merge the source configuration into the target configuration is as follows:
+// The function iterates over every key in the source (src):
+// - If the key does not exist in the target (tgt), it is added.
+// - If the key exists in the target and its value is a map, the function is called recursively.
+// - If the key exists in the target and its value is not a map, it is not merged.
+// Note: A special case occurs if a key is set but its value is nil; it will not be overwritten.
+//
+// --- Example 1 ---
+// === SRC ===
+// api_key: abcdef
+// site: datadoghq.eu
+// proxy:
+//
+//	https: http:proxyserver1
+//
+// === TGT ===
+// api_key:
+// proxy:
+//
+//	http: http:proxyserver2
+//
+// === MERGED ===
+// api_key:
+// site: datadoghq.eu
+// proxy:
+//
+//	https: http:proxyserver1
+//	http: http:proxyserver2
+func (c *safeConfig) AddExtraConfigPaths(ins []string) error {
+	if len(ins) == 0 {
+		return nil
+	}
+	c.Lock()
+	defer c.Unlock()
+	var errs []error
+outerLoop:
+	for _, in := range ins {
+		in, err := filepath.Abs(in)
+		if err != nil {
+			errs = append(errs, fmt.Errorf("could not get absolute path of extra config file '%s': %s", in, err))
+			continue
+		}
```

Comment:
We should try to avoid having a function that returns an error but still modify the object internal.

If we call this with 2 files and one is invalid we still added one to `extraConfigFilePaths`. This is not the most intuitive behavior for the caller that will have to explore the errors to know which file was added to act upon it.

I think we should validate the input entirely and if we have no error add them to `extraConfigFilePaths`. That way the function either succeed or failed but nothing in between,